test/unit/libp2p_port_test.exs

```diff
@@ -86,37 +87,31 @@ defmodule Unit.Libp2pPortTest do
     start_port(:publisher)
     start_port(:gossiper, listen_addr: gossiper_addr)
 
+    # The message sent is the pid
+    message = self() |> :erlang.term_to_binary()
     topic = "/test/gossipping"
-    message = "hello world!"
 
     # Connect the two peers
     %{peer_id: id} = Libp2pPort.get_node_identity(:gossiper)
     :ok = Libp2pPort.add_peer(:publisher, id, gossiper_addr, 999_999_999_999)
 
-    pid = self()
+    # Subscribe to the topic
+    :ok = Libp2pPort.subscribe_to_topic(:gossiper, topic, __MODULE__)
 
     spawn_link(fn ->
-      # Subscribe to the topic
-      :ok = Libp2pPort.subscribe_to_topic(:gossiper, topic)
-      send(pid, :subscribed)
-
-      # Receive the message
-      assert {^topic, message_id, ^message} = Libp2pPort.receive_gossip()
-
-      Libp2pPort.validate_message(message_id, :accept)
-
-      # Send acknowledgement
-      send(pid, :received)
+      # Publish message
+      :ok = Libp2pPort.publish(:publisher, topic, message)
```

Comment:
We don't need the spawn link anymore. This publish can be called right after the subscribe call.